src/utils/UploadBase.js

```diff
@@ -0,0 +1,75 @@
+export default class UploadBase {
+  /**
+   * The abstract base class of a single file uploader.
+   * @abstract
+   * @param file {File | Blob} the file to upload
+   * @param opts {Object} upload options.
+   * @param opts.$rest {Object} an axios instance used for communicating with Girder.
+   * @param opts.parent {Object} upload destination. Must have ``_id`` and ``_modelType``.
+   * @param opts.progress {Function} A progress callback for the upload. It can take an Object
+   *   argument with either ``"indeterminate": true``, or numeric ``current`` fields.
+   */
+  constructor(
```

Comment:
Since this is abstract, it seems like we should omit the constructor altogether. None of the properties assigned here in the constructor are referenced in the rest of the abstract class, meaning they can be pushed down as details of the concrete subclasses. (It's prototypal inheritance so there's no hard and fast right answer, but default/implicit ctor here is what feels right to me.)

Reply:
If we do want to keep this to provide a kind of documentation of what parameters the ctor will receive (which is reasonable), we should not set the `upload`, `offset`, or `behavior` fields in the abstract ctor, as they are not passed to us.

Reply:
Oh, yes. upload, offset, and behavior should not be mentioned in the Base class constructor.